src/dotnet/APIView/APIView/NamedTypeAPIV.cs

```diff
@@ -35,18 +36,24 @@ public class NamedTypeAPIV
         public NamedTypeAPIV(INamedTypeSymbol symbol)
         {
             this.Name = symbol.Name;
-            this.Type = symbol.TypeKind.ToString().ToLower();
+            this.TypeKind = symbol.TypeKind.ToString().ToLower();
             if (symbol.EnumUnderlyingType != null)
-                this.EnumUnderlyingType = symbol.EnumUnderlyingType.ToDisplayString();
+                this.EnumUnderlyingType = new TypeReference(symbol);
             this.Accessibility = symbol.DeclaredAccessibility.ToString().ToLower();
 
-            List<EventAPIV> events = new List<EventAPIV>();
-            List<FieldAPIV> fields = new List<FieldAPIV>();
-            List<string> implementations = new List<string>();
-            List<MethodAPIV> methods = new List<MethodAPIV>();
-            List<NamedTypeAPIV> namedTypes = new List<NamedTypeAPIV>();
-            List<PropertyAPIV> properties = new List<PropertyAPIV>();
-            List<TypeParameterAPIV> typeParameters = new List<TypeParameterAPIV>();
+            var typeParamIndex = symbol.ToDisplayString().LastIndexOf("<");
```

Comment:
Me no likey. Try using https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.inamedtypesymbol.constructedfrom?view=roslyn-dotnet#Microsoft_CodeAnalysis_INamedTypeSymbol_ConstructedFrom instead

Reply:
Also, I would prefer NavigationId generation logic in one place.

Reply:
Agreed, this solution sucks but I hadn't gotten around to replacing it. I'll check it out!